snapcraft/internal/pluginhandler.py

```diff
@@ -656,15 +656,22 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
     for directory in snap_dirs:
         src = os.path.join(srcdir, directory)
         dst = os.path.join(dstdir, directory)
+
+        dir_stat = os.stat(src)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
         os.makedirs(dst, exist_ok=True)
-        shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
+        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
```

Comment:
Excellent point. I think we should.

On Thu, Aug 11, 2016 at 1:39 PM, Kyle Fazzari <notifications@github.com>
wrote:

> In snapcraft/internal/pluginhandler.py
> <https://github.com/snapcore/snapcraft/pull/721#discussion_r74467342>:
>
> >          os.makedirs(dst, exist_ok=True)
> > -        shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
> > +        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
>
> So now instead of copying the mode we're copying the owner info. Which is
> fine. But is there any chance we want to do both?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapcraft/pull/721/files/fbb7f36246cf68d303dc2dd1d6e3def89b700f86#r74467342>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnbyAqKo3gR8ynwhhAes0j0N3CUx9oBks5qe14_gaJpZM4JiW9o>
> .
>